juntagrico/admins/member_admin.py

```diff
@@ -31,3 +42,63 @@ def impersonate_job(self, request, queryset):
 
     impersonate_job.short_description = Config.vocabulary(
         'member') + ' imitieren (impersonate)...'
+
+    def subscription_link(self, obj):
+        return self._get_single_link(obj.subscription, 'juntagrico_subscription_change') \
+               or _('Kein/e/n {}').format(Config.vocabulary('subscription'))
+    subscription_link.short_description = Config.vocabulary('subscription')
+
+    def future_subscription_link(self, obj):
+        return self._get_single_link(obj.future_subscription, 'juntagrico_subscription_change') \
+               or _('Kein/e/n zukünftige/n/s {}').format(Config.vocabulary('subscription'))
+    future_subscription_link.short_description = _('Zukünftige/r/s {}').format(Config.vocabulary('subscription'))
+
+    def old_subscription_link(self, obj):
+        return self._get_multi_link(obj.old_subscriptions.all(), 'juntagrico_subscription_change') \
+               or _('Keine alten {}').format(Config.vocabulary('subscription_pl'))
+    old_subscription_link.short_description = _('Alte {}').format(Config.vocabulary('subscription_pl'))
+
+    def user_link(self, obj):
+        return self._get_single_link(obj.user, 'auth_user_change')
+    user_link.short_description = _('User')
+
+    @staticmethod
+    def _get_single_link(obj, target):
+        return mark_safe('<a href="{}">{}</a>'.format(
+            reverse(f"admin:{target}", args=(obj.pk,)), obj
+        )) if obj else ''
+
+    @staticmethod
+    def _get_multi_link(objs, target):
+        return mark_safe('<br>'.join(['<a href="{}">{}</a>'.format(
+            reverse(f"admin:{target}", args=(obj.pk,)), obj
+        ) for obj in objs]))
+
+
+class MemberAdminWithShares(MemberAdmin):
+    def __init__(self, model, admin_site):
+        super().__init__(model, admin_site)
+        self.readonly_fields.append('share_link')
+        self.fieldsets.insert(-1, (Config.vocabulary('share_pl'), {'fields': ['share_link']}))
+
+    def share_link(self, obj):
+        return self._get_multi_link(obj.share_set.all(), 'juntagrico_share_change') \
+               or _('Kein/e/n {}').format(Config.vocabulary('share'))
+    share_link.short_description = Config.vocabulary('share_pl')
+
+
+class JuntagricoUserAdmin(UserAdmin):
+    readonly_fields = ('member_link',)
+    list_display = ('username', 'member', 'is_staff')
+
+    def __init__(self, model, admin_site):
+        super().__init__(model, admin_site)
+        self.fieldsets = (self.fieldsets[0], *self.fieldsets[2:])  # remove irrelevant section
+        self.fieldsets[0][1]['fields'] += ('member_link',)  # add link to member
```

Comment:
da wir nicht wissen ob das fieldset 0 immer daselbe sein wird sollten wir unsern member link in ein neues fieldset packen am besten ganz an anfag oder am ende